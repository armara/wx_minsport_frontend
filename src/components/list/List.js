import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import Image from 'components/image/Image'

import 'components/list/List.scss'

const rowRenderer = item => {
  if (item.type === 'hr') {
    return <hr />
  }

  const { spanClassName } = item

  return (
    <>
      <div>
        {item.image && <Image src={item.image} />}
        <h4>{item.field}</h4>
      </div>

      {!item.value ? (
        '---'
      ) : (
        <span className={spanClassName}>{item.value}</span>
      )}
    </>
  )
}

const List = ({
  id,
  keyId,
  items,
  children,
  className,
  rowClassName,
  getRowClassName,
  notFoundItem,
  childrenProps,
}) => (
  <ul className={classnames([className, 'base-list'])} id={id}>
    {items.length === 0 ? (
      <li className="base-list_not-found">{notFoundItem}</li>
    ) : (
      items.map((item, i) => (
        <li
          key={item[keyId]}
          className={classnames([rowClassName, getRowClassName(item, i)])}
        >
          {children ? children(item, childrenProps) : rowRenderer(item)}
        </li>
      ))
    )}
  </ul>
)

List.defaultProps = {
  id: null,
  keyId: 'key',
  className: '',
  rowClassName: '',
  children: '',
  getRowClassName: () => {},
  childrenProps: {},
  notFoundItem: 'Не найдено',
}

List.propTypes = {
  id: PropTypes.string,
  keyId: PropTypes.string,
  className: PropTypes.string,
  rowClassName: PropTypes.string,
  getRowClassName: PropTypes.func,
  notFoundItem: PropTypes.string,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      image: PropTypes.string,
      field: PropTypes.string,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      type: PropTypes.oneOf(['hr']),
    })
  ).isRequired,
  childrenProps: PropTypes.shape({}),
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
}

export default List
