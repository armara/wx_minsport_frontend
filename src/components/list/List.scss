@import "styles/mixins";

.base-list {
  .base-list_not-found {
    font-style: italic;
  }

  hr {
    width: 100%;
    margin-top: 15px;
    margin-bottom: 15px;
  }

  li + li {
    margin-top: 15px;
  }

  li {
    display: flex;
    justify-content: space-between;

    h4 {
      @include roboto-regular-text($small, 18px, $evil-gray);
      display: flex;
      align-items: center;
    }

    & > div {
      display: flex;

      img {
        margin-right: 10px;
      }
    }

    & > span {
      @include nav-header-text($small, '18px', $black);
      display: flex;
      align-items: center;
    }
  }
}

.tabs-list {
  padding-top: 0;

  li {
    height: 45px;
  }

  li + li {
    margin-top: 0;
  }
}