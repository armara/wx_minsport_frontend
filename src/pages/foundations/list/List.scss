@import "styles/mixins";

.foundations-list {
  height: 100%;
  display: flex;
  flex-flow: column;

  .foundations-list_filters {
    display: flex;
    align-items: center;
    padding-top: 25px;
    padding-bottom: 25px;
  }

  .foundations-list_filters_dropdown + .foundations-list_filters_dropdown {
    margin-left: 11px;
  }

  .foundations-list_filters_types-trigger,
  .foundations-list_filters_types-menu {
    min-width: 220px;
  }

  .foundations-list_filters_types-trigger,
  .foundations-list_filters_trigger {
    button.is-white {
      @include nav-header-text($small, '18px', $evil-gray);
      width: 100%;
    }
  }

  .foundations-list_filters_trigger,
  .foundations-list_filters_menu {
    min-width: 200px;
  }

  .foundations-list_search-input {
    margin-left: auto;
  }

  .foundations-list_table {
    flex-grow: 1;
    align-items: flex-start;
    margin-bottom: 20px;
  }
}