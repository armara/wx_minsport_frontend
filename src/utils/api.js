import axios from 'axios'
import config from 'config'

import { getUserWorkplaceId } from 'utils/user'

const facilityId = getUserWorkplaceId()
const {
  BACKEND_URL,
  auth: { username, password },
  service: {
    auth,
    sports,
    districts,
    facilitiesV2,
    federations,
    organizations,
    personsV3,
    scheduleV2,
  },
} = config

const authUrl = BACKEND_URL + auth
const facilitiesv2Url = BACKEND_URL + facilitiesV2
const districtsUrl = BACKEND_URL + districts
const sportsUrl = BACKEND_URL + sports
const scheduleV2Url = BACKEND_URL + scheduleV2
const federationUrl = BACKEND_URL + federations
const personsV3Url = BACKEND_URL + personsV3
const organizationsUrl = BACKEND_URL + organizations

const axiosAuth = axios.create({
  baseURL: authUrl,
  auth: {
    username,
    password,
  },
})

export default () => ({
  auth: formData => axiosAuth.post(`/token`, formData),

  getOrganizationById: id =>
    axios.get(`${organizationsUrl}/organizations/${id}`),
  getFederationById: id => axios.get(`${federationUrl}/federation/${id}`),

  postGroup: body => axios.post(`${personsV3Url}/group`, { ...body }),
  putGroup: (body, id) => axios.put(`${personsV3Url}/group/${id}`, { ...body }),
  deleteGroup: (id, body) =>
    axios.delete(`${personsV3Url}/group/${id}`, {
      data: body,
      headers: {
        'Content-Type': 'application/json',
      },
    }),

  getDistricts: () => axios.get(`${districtsUrl}/district/all`),
  getSubdistricts: districtId =>
    axios.get(`${districtsUrl}/subdistrict/allByDistrict/`, {
      params: {
        districtId,
      },
    }),

  getGroups: query => axios.get(`${personsV3Url}/group${query}`),
  getPersons: query => axios.get(`${personsV3Url}/person${query}`),
  updatePerson: (personId, body) =>
    axios.put(`${personsV3Url}/person/${personId}`, { ...body }),
  getGroupById: id => axios.get(`${personsV3Url}/group/${id}`),
  getPersonById: id => axios.get(`${personsV3Url}/person/${id}`),
  getPositions: () => axios.get(`${personsV3Url}/position`),
  addPerson: body => axios.post(`${personsV3Url}/person`, { ...body }),
  removePerson: (id, body) =>
    axios.delete(`${personsV3Url}/person/${id}`, {
      data: body,
      headers: {
        'Content-Type': 'application/json',
      },
    }),
  getFederations: () => axios.get(`${federationUrl}/federations`),
  getPersonAvatarUrl: personId => `${personsV3Url}/person/${personId}/avatar`,

  uploadPersonAvatar2: (personId, formData) =>
    axios.post(`${personsV3Url}/person/${personId}/avatar`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    }),

  getRanks: () => axios.get(`${personsV3Url}/rank`),
  getGroupsByPersonId: (query = '') =>
    axios.get(`${personsV3Url}/group/${query}`),

  getEventTypes: () => axios.get(`${scheduleV2Url}/types/events`),
  addEventItem: body => axios.post(`${scheduleV2Url}/events`, { ...body }),

  getScheduleEventById: id => axios.get(`${scheduleV2Url}/events/${id}`),
  getScheduleById: id => axios.get(`${scheduleV2Url}/schedules/${id}`),
  editScheduleItem: (body, id) =>
    axios.put(`${scheduleV2Url}/schedules/${id}`, { ...body }),

  getFacilitySports: _facilityId =>
    axios.get(`${sportsUrl}/fsports/`, {
      params: {
        facilityId: _facilityId || facilityId,
        pageSize: 10000,
        returnDisciplines: true,
      },
    }),
  getFacilityDisciplines: () =>
    axios.get(
      `${sportsUrl}/fdisciplines/
      `,
      {
        params: {
          facilityId,
          pageSize: 10000,
        },
      }
    ),
  getSportsBulk: ids => axios.post(`${sportsUrl}/sports/bulk`, { ids }),
  getSportBy: id => axios.get(`${sportsUrl}/sports/${id}`),
  getFacilitySportBy: id => axios.get(`${sportsUrl}/fsports/${id}`),
  getFullSports: (params = {}) =>
    axios.get(`${sportsUrl}/sports/`, {
      params,
    }),
  getFullDisciplines: () => axios.get(`${sportsUrl}/disciplines`),

  getAllFacilities: () => axios.get(`${facilitiesv2Url}/facility/all`),
  getAreas: () => axios.get(`${facilitiesv2Url}/facility/${facilityId}/areas`),
  getAbonnements: abonnementId =>
    axios.get(`${facilitiesv2Url}/facility/abonnement/${abonnementId}`),
  getAllAbonnements: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/abonnements`),
  getZones: areaId =>
    axios.get(`${facilitiesv2Url}/facility/area/${areaId}/zones`),
  getFacilityByIdSummary: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/summary`),
  getFacilityById: id => axios.get(`${facilitiesv2Url}/facility/${id}`),
  getOrgTypes: () => axios.get(`${organizationsUrl}/organizations/types`),
  getAllOrganizations: () => axios.get(`${organizationsUrl}/organizations`),
  postOrganization: body =>
    axios.post(`${organizationsUrl}/organizations`, body),
  getOrganizationByQuery: params =>
    axios.get(`${organizationsUrl}/organizations/`, { params }),
  getContractTypes: () =>
    axios.get(`${facilitiesv2Url}/facility/contract/type/all`),
  getContractFiles: contractId =>
    axios.get(`${facilitiesv2Url}/facility/contract/${contractId}/files`),
  getZonesByContractId: contractId =>
    axios.get(`${facilitiesv2Url}/facility/contract/${contractId}/zones`),
  postContract: body =>
    axios.post(`${facilitiesv2Url}/facility/contract`, body),
  putContract: contract =>
    axios.put(`${facilitiesv2Url}/facility/contract`, contract),
  postZone: zone =>
    axios.post(`${facilitiesv2Url}/facility/contract/zone`, zone),
  removeZones: (contractId, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/${contractId}/zones`, {
      data: { reason },
    }),
  getAllContracts: () =>
    axios.get(`${facilitiesv2Url}/facility/${facilityId}/contracts`),
  getAllOrganizationContracts: orgId =>
    axios.get(`${facilitiesv2Url}/facility/contracts?orgId=${orgId}`),
  getContract: id => axios.get(`${facilitiesv2Url}/facility/contract/${id}`),
  removeContract: (contractId, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/${contractId}`, {
      data: { reason },
    }),
  postFile: file =>
    axios.post(`${facilitiesv2Url}/facility/contract/file`, file),
  removeFile: (id, reason = '') =>
    axios.delete(`${facilitiesv2Url}/facility/contract/file/${id}`, {
      data: { reason },
    }),

  getEvents: query => axios.get(`${scheduleV2Url}/events${query}`),
  getSchedules: query =>
    axios.get(`${scheduleV2Url}/schedules${query}`, {
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    }),

  removeSchedule: (id, body) =>
    axios.delete(`${scheduleV2Url}/schedules/${id}`, {
      data: body,
    }),

  removeEvent: (id, body) =>
    axios.delete(`${scheduleV2Url}/events/${id}`, {
      data: body,
    }),
})

async function requestIfIdDefined(request, id) {
  let response = null
  try {
    if (id) {
      const { data } = await request(id)
      response = data
    }
    /* eslint-disable-next-line */
  } catch {}
  return response
}

export { requestIfIdDefined }
