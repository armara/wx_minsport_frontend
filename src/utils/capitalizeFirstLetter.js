export default string => string.replace(string[0], string[0].toUpperCase())
