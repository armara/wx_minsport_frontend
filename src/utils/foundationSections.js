import editorImage from 'assets/images/placeholder.jpg'

export const FoundationItems = ({
  legalData,
  orgType,
  organizationName,
  tin,
  description,
  contractTitle,
  sportTitle,
  reg,
  end,
}) => [
  {
    field: 'Название контракта',
    value: legalData,
    key: 'contract-legal-data',
  },
  {
    field: 'Описание контракта',
    value: description,
    key: 'contract-description',
  },
  {
    field: 'Название организации',
    value: organizationName,
    key: 'organization-name',
  },
  {
    field: 'Тип организации',
    value: (orgType || {}).name,
    key: 'foundation-type',
  },
  {
    field: 'ИНН',
    value: tin,
    key: 'organization-tin',
  },
  {
    field: 'Вид спорта',
    value: sportTitle,
    key: 'sport-title',
  },
  {
    field: 'Классификатор договора',
    value: contractTitle,
    key: 'contract-classifier',
  },
  {
    field: 'Дата заключения договора',
    value: reg,
    key: 'contract-start-date',
  },
  {
    field: 'Дата окончания договора',
    value: end,
    key: 'contract-expire-date',
  },
  {
    type: 'hr',
    key: 'hr-2',
  },
]

export const FoundationTeamSections = [
  {
    title: 'Спортсмены',
    key: 'sportsmen',
  },
  {
    title: 'Тренеры',
    key: 'coaches',
  },
]

export const FoundationSportsmen = [
  {
    image: editorImage,
    field: 'Ковалев В.А.',
    value: 'Мастер спорта',
    key: 'sportsmen-1',
  },
  {
    image: editorImage,
    field: 'Ромодановский Ф.',
    value: 'Заслуженный мастер спорта',
    key: 'sportsmen-2',
  },
  {
    image: editorImage,
    field: 'Франц Лефорт',
    value: 'Мастер спорта',
    key: 'sportsmen-3',
  },
  {
    image: editorImage,
    field: 'Нарышкин П.А.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-4',
  },
  {
    image: editorImage,
    field: 'Шереметев Б.П.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-5',
  },
  {
    image: editorImage,
    field: 'Штейн К.',
    value: 'Заслуженный мастер спорта',
    key: 'sportsmen-6',
  },
  {
    image: editorImage,
    field: 'Гусин Л.Д.',
    value: 'Мастер спорта',
    key: 'sportsmen-7',
  },
  {
    image: editorImage,
    field: 'Плещеев И.А.',
    value: 'Мастер спорта международного класса',
    key: 'sportsmen-8',
  },
]

export const FoundationFiles = [
  {
    name: 'Договор',
    key: 'contract',
  },
  {
    name: 'Приложение к договору',
    key: 'contract-addon',
  },
]
