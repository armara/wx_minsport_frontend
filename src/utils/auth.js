import axios from 'axios'
import debounce from 'lodash.debounce'

import { getUserData, createWorkplace, updateUserData } from 'utils/user'
import api from 'utils/api'
import history from 'utils/utils'
import dayjs, { hasTimePassed } from 'utils/day'

const MILLISECONDS_BEFORE_REFRESH_REQUEST = 1000

export const isAuthenticated = () => Object.keys(getUserData()).length > 0

const getUserDataFromResponse = response => {
  const {
    data: {
      access_token: accessToken,
      email,
      jti,
      phone,
      facilities,
      federations,
      organizations,
      ministries,
      first_name: firstName,
      sur_name: surName,
      patronymic_name: patronymicName,
      expires_in: expiresIn,
      refresh_token: refreshToken,
      token_type: tokenType,
    } = {},
  } = response

  const facilitiesIds = Object.keys(facilities)
  const federationsIds = Object.keys(federations)
  const organizationsIds = Object.keys(organizations)
  const ministriesIds = Object.keys(ministries)

  return {
    accessToken,
    email,
    jti,
    firstName: firstName || '',
    surName: surName || '',
    patronymicName: patronymicName || '',
    phone,
    facilities: facilitiesIds,
    federations: federationsIds,
    organizations: organizationsIds,
    ministries: ministriesIds,
    expiresIn,
    refreshToken,
    tokenType,
  }
}

const chooseInitialUserWorkplace = ({
  facilities,
  organizations,
  federations,
  ministries,
}) => {
  let type
  let id
  if (facilities.length > 0) {
    type = 'facilities'
    ;[id] = facilities
  } else if (organizations.length > 0) {
    type = 'organizations'
    ;[id] = organizations
  } else if (federations.length > 0) {
    type = 'federations'
    ;[id] = federations
  } else if (ministries.length > 0) {
    type = 'ministries'
    ;[id] = ministries
  }

  return createWorkplace(type, id)
}

export const login = async ({ username, password }) => {
  const formData = new FormData()
  formData.append('grant_type', 'password')
  formData.append('username', username) // petr007
  formData.append('password', password) // qwerty

  return (
    api()
      .auth(formData)
      .then(response => {
        if (response.status === 200) {
          const userData = getUserDataFromResponse(response)
          const workPlace = chooseInitialUserWorkplace(userData)
          const time = dayjs().toISOString()
          updateUserData(userData, { workPlace, time })
        }

        return response
      })
      // eslint-disable-next-line no-console
      .catch(console.error)
  )
}

export const logout = () => {
  localStorage.removeItem('user')
}

const refreshAccessToken = refreshToken => {
  const formData = new FormData()
  formData.append('grant_type', 'refresh_token')
  formData.append('refresh_token', refreshToken)

  return api()
    .auth(formData)
    .then(response => {
      if (response.status !== 200) {
        logout()
        window.location.assign('/login')
      }

      const currentUserData = getUserData()
      const nextUserData = getUserDataFromResponse(response)
      const time = dayjs().toISOString()
      updateUserData(currentUserData, { ...nextUserData, time })

      return response
    })
    .catch(() => {
      logout()
      window.location.assign('/login')
    })
}

const debouncedRefreshAccessToken = debounce(
  refreshAccessToken,
  MILLISECONDS_BEFORE_REFRESH_REQUEST,
  {
    leading: true,
    trailing: false,
  }
)

const getValidUserToken = async () => {
  const { time, expiresIn, refreshToken } = getUserData()

  const isPassed = hasTimePassed(time, expiresIn)
  if (isPassed) {
    await debouncedRefreshAccessToken(refreshToken)
    debouncedRefreshAccessToken.cancel()
  }

  return getUserData()
}

export const setupInterceptor = () => {
  axios.interceptors.response.use(
    response => {
      return response
    },
    error => {
      return Promise.reject(error)
    }
  )

  axios.interceptors.request.use(
    async config => {
      const isAuthed = isAuthenticated()

      if (config.url.endsWith('/login')) {
        return config
      }

      if (!isAuthed) {
        history.push('/login')
        return config
      }

      const { accessToken, tokenType } = await getValidUserToken()
      return {
        ...config,
        headers: {
          Authorization: `${tokenType} ${accessToken}`,
        },
      }
    },
    error => {
      return Promise.reject(error)
    }
  )
}

export const facilitiesPrefix = 'facilities'
export const organizationsPrefix = 'organizations'
