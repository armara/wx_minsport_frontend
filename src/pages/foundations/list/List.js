import React, { useState, useEffect, useCallback } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import debounce from 'lodash.debounce'

import TableCellActions from 'components/table/tableCellActions/TableCellActions'
import TableArrowPaginator from 'components/table/tableArrowPaginator/TableArrowPaginator'
import TableCellFoundation from 'components/table/tableCellIcon/TableCellIcon'
import TableCellText from 'components/table/tableCellText/TableCellText'
import Table from 'components/table/Table'
import AddFoundationButton from 'pages/foundations/components/addFoundationButton/AddFoundationButton'
import FilterPanel from 'components/filterPanel/FilterPanel'

import history from 'utils/utils'
import { facilitiesPrefix as prefix } from 'utils/auth'
import {
  setModal,
  setCurrentContract,
} from 'store/actions/foundation/foundation'
import {
  getAllContracts,
  getContracts,
  searchByName,
  filterBySport,
  filterByType,
} from 'store/actionCreators/foundation/foundation'
import {
  foundationAllowedSportsSelector,
  foundationAllowedTypesSelector,
  foundationCurrentContractsSelector,
  foundationCurrentPageSelector,
  foundationLoadingSelector,
  foundationPageSizeSelector,
  foundationPagesSelector,
  foundationSportFilterSelector,
  foundationTotalSelector,
  foundationTypeFilterSelector,
} from 'store/selectors/foundation'

import './List.scss'

const columnsCallback = (onEditClick, onDeleteClick) => [
  {
    id: 'Юридическое название договора',
    Header: 'Юридическое название договора',
    isFilterable: true,
    accessor: row => <TableCellFoundation foundation={row.legalData} />,
  },
  {
    id: 'Тип договора',
    Header: 'Тип договора',
    isFilterable: true,
    accessor: row => <TableCellText text={row.contractTitle} />,
  },
  {
    id: 'Заказчик',
    Header: 'Заказчик',
    isFilterable: true,
    accessor: row => <TableCellText text={row.organizationName} />,
  },
  {
    id: 'tin',
    Header: 'ИНН',
    isFilterable: true,
    accessor: row => <TableCellText text={row.organizationTin} />,
  },
  {
    id: 'Вид спорта',
    Header: 'Вид спорта',
    isFilterable: true,
    accessor: row => <TableCellText text={row.sportTitle} />,
  },
  {
    id: 'Регистрация',
    Header: 'Регистрация',
    isFilterable: true,
    accessor: row => <TableCellText text={row.reg} />,
  },
  {
    id: 'Действия',
    Header: ' ',
    width: 80,
    isFilterable: false,
    accessor: row => (
      <TableCellActions
        onEditClick={event => onEditClick(event, row.id)}
        onDeleteClick={event => onDeleteClick(event, row.id)}
      />
    ),
  },
]

function List({
  setModalState,
  setContract,
  currentPage,
  pageSize,
  pages,
  filterTypeValue,
  contracts,
  total,
  sportFilterValue,
  fetchContracts,
  loading,
  allowedSports,
  allowedTypes,
  handleSportChange,
  handleTypeChange,
  handleNameChange,
  fetchPage,
}) {
  const [searchText, setSearchText] = useState('')

  const onEditClick = useCallback((event, id) => {
    event.stopPropagation()
    history.push(`/${prefix}/foundations/edit/${id}`)
  }, [])

  const onDeleteClick = useCallback(
    (event, id) => {
      event.stopPropagation()
      setContract({ id })
      setModalState(true)
    },
    [setContract, setModalState]
  )

  const columns = columnsCallback(onEditClick, onDeleteClick)

  useEffect(() => {
    fetchContracts()
  }, [fetchContracts])

  const handleOnBackClick = () => {
    if (currentPage !== 0) {
      fetchPage(currentPage - 1)
    }
  }

  const handleOnForwardClick = () => {
    if (currentPage !== pages) {
      fetchPage(currentPage + 1)
    }
  }

  const debouncedOnChange = useCallback(
    debounce(value => handleNameChange(value), 200),
    []
  )

  const onChange = event => {
    setSearchText(event.target.value)
    debouncedOnChange(event.target.value)
  }

  return (
    <div className="foundations-list">
      <FilterPanel
        onSearch={onChange}
        filters={[
          {
            id: 1,
            items: allowedTypes,
            selectedItem: filterTypeValue,
            onSelect: handleTypeChange,
            inputPlaceholder: 'Выбрать тип организации',
          },
          {
            id: 2,
            items: allowedSports,
            onSelect: handleSportChange,
            inputPlaceholder: 'Выбрать вид спорта',
            selectedItem: sportFilterValue,
          },
        ]}
        searchPlaceholder="Поиск договоров"
        searchValue={searchText}
      />

      <Table
        columns={columns}
        data={contracts}
        loading={loading}
        onRowClick={rowInfo =>
          history.push(`/${prefix}/foundations/${rowInfo.original.id}`)
        }
      />

      <TableArrowPaginator
        totalRows={total}
        currentPage={currentPage}
        onBackClick={handleOnBackClick}
        onForwardClick={handleOnForwardClick}
        pageSize={pageSize}
        pages={pages}
      >
        <AddFoundationButton />
      </TableArrowPaginator>
    </div>
  )
}

List.propTypes = {
  contracts: PropTypes.arrayOf(PropTypes.object).isRequired,
  fetchPage: PropTypes.func.isRequired,
  fetchContracts: PropTypes.func.isRequired,
  handleSportChange: PropTypes.func.isRequired,
  handleTypeChange: PropTypes.func.isRequired,
  handleNameChange: PropTypes.func.isRequired,
  total: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  pages: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  loading: PropTypes.bool.isRequired,
  filterTypeValue: PropTypes.shape({}).isRequired,
  sportFilterValue: PropTypes.shape({}).isRequired,
  allowedSports: PropTypes.arrayOf(PropTypes.object).isRequired,
  allowedTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
  setContract: PropTypes.func.isRequired,
  setModalState: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  contracts: foundationCurrentContractsSelector(state),
  total: foundationTotalSelector(state),
  pageSize: foundationPageSizeSelector(state),
  pages: foundationPagesSelector(state),
  currentPage: foundationCurrentPageSelector(state),
  loading: foundationLoadingSelector(state),
  allowedSports: foundationAllowedSportsSelector(state),
  allowedTypes: foundationAllowedTypesSelector(state),
  filterTypeValue: foundationTypeFilterSelector(state),
  sportFilterValue: foundationSportFilterSelector(state),
})

const mapDispatchToProps = dispatch => ({
  setContract: id => dispatch(setCurrentContract(id)),
  setModalState: isModalOpen => dispatch(setModal(isModalOpen)),
  fetchPage: page => dispatch(getContracts(page)),
  fetchContracts: () => dispatch(getAllContracts()),
  handleSportChange: item => dispatch(filterBySport(item)),
  handleTypeChange: item => dispatch(filterByType(item)),
  handleNameChange: item => dispatch(searchByName(item)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(List)
