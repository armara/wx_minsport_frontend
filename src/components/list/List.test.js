import React from 'react'
import List from 'components/list/List'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import { FoundationItems } from 'utils/foundationSections'

configure({ adapter: new Adapter() })

describe('<List />', () => {
  it('renders correctly default props', () => {
    const tree = renderer.create(<List items={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly default mock list', () => {
    const tree = renderer
      .create(
        <List
          items={FoundationItems({
            orgType: 'orgType1',
            contractTitle: 'contractTitle1',
            reg: 0,
            end: 10,
          })}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
