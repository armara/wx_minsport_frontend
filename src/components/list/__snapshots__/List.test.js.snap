// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders correctly default mock list 1`] = `
<ul
  className="base-list"
  id={null}
>
  <li
    className=""
  >
    <div>
      <h4>
        Название контракта
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Описание контракта
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Название организации
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Тип организации
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        ИНН
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Вид спорта
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Классификатор договора
      </h4>
    </div>
    <span>
      contractTitle1
    </span>
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Дата заключения договора
      </h4>
    </div>
    ---
  </li>
  <li
    className=""
  >
    <div>
      <h4>
        Дата окончания договора
      </h4>
    </div>
    <span>
      10
    </span>
  </li>
  <li
    className=""
  >
    <hr />
  </li>
</ul>
`;

exports[`<List /> renders correctly default props 1`] = `
<ul
  className="base-list"
  id={null}
>
  <li
    className="base-list_not-found"
  >
    Не найдено
  </li>
</ul>
`;
